src/webgl/material.js

```diff
@@ -513,6 +513,65 @@ function material(p5, fn){
     p5._validateParameters('createShader', arguments);
     return new Shader(this._renderer, vertSrc, fragSrc, options);
   };
+  /**
+   * Creates and loads a filter shader from an external file.
+   *
+   * @method loadFilterShader
+   * @param {String} fragFilename path to the fragment shader file
+   * @param {Function} [successCallback] callback to be called once the shader is
+   *                                     loaded. Will be passed the
+   *                                     <a href="#/p5.Shader">p5.Shader</a> object.
+   * @param {Function} [failureCallback] callback to be called if there is an error
+   *                                     loading the shader. Will be passed the
+   *                                     error event.
+   * @return {Promise<p5.Shader>} a promise that resolves with a shader object
```

Comment:
This looks good @Rishab87 !

@limzykenneth we probably need to update the docs for other `load*` methods to indicate they return promises now if we haven't already, and likely we will need to update the FES code to handle templated Promise types

Reply:
For most of the load functions I did do a round of simple documentation update so most should be fine